src/components/Api.js

```diff
@@ -0,0 +1,98 @@
+export default class Api {
+  constructor({ headers, baseUrl }) {
+    this._headers = headers;
+    this._baseUrl = baseUrl;
+  }
+
+  _checkResponse(res) {
+    if (res.ok) {
+      return res.json();
+    }
+    return Promise.reject(`Error: ${res.status}`);
+  }
+
+  getUserInfo() {
+    return fetch(`${this._baseUrl}/users/me`, {
+      headers: this._headers
+    }).then(this._checkResponse)
```

Comment:
**Could be improved**

You can make a special method (function ) for fetching and checking responses **not to duplicate** it in every request:

```js

  _request(url, options) {
    return fetch(url, options).then(this._checkResponse)
  }

```

And now you can safely replace `fetch` with `this._request` and remove repeated checking. And you don’t need to change anything else.

Reply:
Thank you for the suggestion. I'm a little confused on how I would include options?

Reply:
> Thank you for the suggestion. I'm a little confused on how I would include options?

Everything will be the same. Only `.then(this._checkResponse);` should be removed from all requests

```js


  _request(url, options) {
    return fetch(url, options).then(this._checkResponse);
  }

  getUserInfo() {
    return this._request(`${this._baseUrl}/users/me`, {  // the options are the same
      headers: this._headers,
    });
  }

```